cranelift/codegen/src/opts/algebraic.isle

```diff
@@ -520,14 +468,6 @@
        (select ty (fcmp _ (FloatCC.GreaterThan) x y) x y))
       (fmax_pseudo ty x y))
 
-;; Do the same for vectors
-(rule (simplify
-       (vselect ty (fcmp _ (FloatCC.LessThan) x y) x y))
-      (fmin_pseudo ty x y))
-(rule (simplify
-       (vselect ty (fcmp _ (FloatCC.GreaterThan) x y) x y))
-      (fmax_pseudo ty x y))
```

Comment:
I think replacing `vselect` with `bitselect` almost works for `fcmp` as well, except that typesets containing vectors of floats turn into vectors of ints of the same size, but `bitselect` requires all the types to be the same.

Maybe we should extend `bitselect` to accept int vectors of the right width even for selecting between float vectors? I think that would remove the need for this PR's changes to NaN canonicalization, and probably be more useful in general.

Reply:
Would you be ok with a follow-up issue for this?

Reply:
Sure, assuming the changes I've just suggested to NaN canonicalization seem reasonable to you, we can think about `bitselect` semantics in more detail later. If we see a lot more cases of `fcmp`->`bitcast`->`bitselect` I might prioritize it higher.

For the record, I'm not concerned about losing these two specific optimization rules, since they weren't motivated by seeing this pattern in any benchmarks; we just added them because we could. But it'd be nice if you'd leave a comment behind mentioning something like "TODO: this optimization also applies to vectors but the pattern to match is more complicated".

Reply:
Ok sounds good. I mostly want to not pile up too much on this PR and I'm not too familiar with cranelift's instruction format builde, so I'd prefer to set that aside for another time rather than doing it here-and-now. I do think it makes sense, though that `bitselect` always takes an integer mask. Most of these optimizations probably won't fire from wasm code though since wasm `v128.bitselect` forces everything through `i8x16` which loses most of the type information. That being said these optimizations aren't all that important for wasm probably since if you're using SIMD you're probably using LLVM which would already have done optimizations like this ideally.

Reply:
I've opened https://github.com/bytecodealliance/wasmtime/issues/5962 for this